Nostrid.Core/Data/DmService.cs

```diff
@@ -89,7 +89,19 @@ public void SetLastRead(string accountId, string otherAccountId)
         var accountIsL = accountId.CompareTo(otherAccountId) < 0;
 
         var now = DateTime.UtcNow;
-        if (accountIsL)
+
+        if (accountId == otherAccountId)
+        {
+            // you can dm yourself, this fixes the message counts.
+            db.DmPairs
+                .Where(p => p.AccountL == accountId && p.AccountH == otherAccountId)
+                .ExecuteUpdate(p => p.SetProperty(p => p.LastReadL, now));
+
+            db.DmPairs
+                .Where(p => p.AccountH == accountId && p.AccountL == otherAccountId)
+                .ExecuteUpdate(p => p.SetProperty(p => p.LastReadH, now));
```

Comment:
I think these two statements can be combined into one. Can you give it a try?

Reply:
Yeah was thinking that myself, will take a look.

Reply:
Do we even need to report self messages as unread, seems pointless to even bother, if you send yourself a message you have already seen it.

Reply:
Yes, I believe it doesn't make sense to notify self messages